pkg/sqlstore/impl/system/store.go

```diff
@@ -79,6 +82,11 @@ func New(dbURI string, chainID tableland.ChainID) (*SystemStore, error) {
 	return systemStore, nil
 }
 
+// SetReadResolver sets the resolver for read queries.
+func (s *SystemStore) SetReadResolver(resolver sqlparser.ReadStatementResolver) {
```

Comment:
Cannot pass this at the constructor level because of cyclical dependency between the event processor and the `SystemStore`, so i created a setter